src/components/vm/consoles/vncBody.jsx

```diff
@@ -17,47 +17,83 @@
  * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
  */
 
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
-import { FormGroup, Grid, GridItem, TextInput } from "@patternfly/react-core";
+import {
+    FormGroup, FormHelperText, HelperText, HelperTextItem,
+    InputGroup, TextInput, Button
+} from "@patternfly/react-core";
+import { EyeIcon, EyeSlashIcon } from "@patternfly/react-icons";
 
 import cockpit from 'cockpit';
 
 const _ = cockpit.gettext;
 
-export const VncRow = ({ idPrefix, onValueChanged, dialogValues }) => {
+export const VncRow = ({ idPrefix, onValueChanged, dialogValues, validationErrors }) => {
+    const [showPassword, setShowPassword] = useState(false);
+
     return (
-        <Grid hasGutter md={6}>
-            <GridItem span={6}>
-                <FormGroup fieldId={`${idPrefix}-address`} label={_("VNC address")}>
-                    <TextInput id={`${idPrefix}-address`}
-                           value={dialogValues.vncAddress}
-                           type="text"
-                           onChange={(event) => onValueChanged('vncAddress', event.target.value)} />
-                </FormGroup>
-            </GridItem>
-            <GridItem span={6}>
-                <FormGroup fieldId={`${idPrefix}-port`} label={_("VNC port")}>
-                    <TextInput id={`${idPrefix}-port`}
-                           value={dialogValues.vncPort}
-                           type="number"
-                           onChange={(event) => onValueChanged('vncPort', event.target.value)} />
-                </FormGroup>
-            </GridItem>
-            <GridItem span={6}>
-                <FormGroup fieldId={`${idPrefix}-password`} label={_("VNC password")}>
-                    <TextInput id={`${idPrefix}-password`}
-                           value={dialogValues.vncPassword}
-                           type="password"
-                           onChange={(event) => onValueChanged('vncPassword', event.target.value)} />
-                </FormGroup>
-            </GridItem>
-        </Grid>
+        <>
+            <Grid hasGutter md={6}>
+                <GridItem span={6}>
+                    <FormGroup fieldId={`${idPrefix}-address`} label={_("Listening address")}>
```

Comment:
The "Listening" is redundant here, a bit jargony, and makes the label unnecessarily wide (translations will only aggravate this). The address wasn't part of your [recent round of screenshots](https://github.com/cockpit-project/cockpit-machines/pull/1973#issuecomment-2615732002) but of course we need it.

Reply:
Address is out of scope now.